internal/pkg/vm/actor/builtin/initactor/init.go

```diff
@@ -20,29 +22,41 @@ type State struct {
 	Network string
 }
 
+const (
+	GetNetwork types.MethodID = iota
+)
+
+// NewActor returns a init actor.
+func NewActor() *actor.Actor {
+	return actor.NewActor(types.InitActorCodeCid, types.ZeroAttoFIL)
+}
+
+//
+// ExecutableActor impl for Actor
+//
+
 // Ensure InitActor is an ExecutableActor at compile time.
 var _ exec.ExecutableActor = (*Actor)(nil)
 
-// initExports are the publicly (externally callable) methods of the AccountActor.
-var initExports = exec.Exports{
-	"getNetwork": &exec.FunctionSignature{
+var signatures = exec.Exports{
+	GetNetwork: &exec.FunctionSignature{
 		Params: []abi.Type{},
 		Return: []abi.Type{abi.String},
 	},
 }
 
-// Exports makes the available methods for this contract available.
-func (a *Actor) Exports() exec.Exports {
-	return initExports
-}
-
-// NewActor returns a init actor.
-func NewActor() *actor.Actor {
-	return actor.NewActor(types.InitActorCodeCid, types.ZeroAttoFIL)
+// Method returns method definition for a given method id.
+func (a *Actor) Method(id types.MethodID) (exec.Method, *exec.FunctionSignature, bool) {
+	switch id {
+	case GetNetwork:
+		return reflect.ValueOf((*Impl)(a).GetNetwork), signatures[GetNetwork], true
```

Comment:
Reflection is heavy overhead for this dispatch. It wasn't necessary in the existing code, and this is a good opportunity to fix it. It would be more direct (both understandable and in execution) to look up an array indexed by method id. In fact, something very similar to the existing Exports list (but including the method pointer) would probably work great.

If you're planning to replace this mechanism in a follow-up, 👌 .

Reply:
Yes, I want all of this to go away during the `VMContext to spec` and `method execution up to spec` tasks. Out-of-scope to cleanup on this PR.